OpenEdXMobile/src/test/java/org/edx/mobile/view/BaseVideosDownloadStateActivityTest.java

```diff
@@ -1,110 +0,0 @@
-package org.edx.mobile.view;
-
-import android.view.Menu;
-import android.view.MenuItem;
-
-import org.edx.mobile.R;
-import org.edx.mobile.base.BaseVideosDownloadStateActivity;
-import org.edx.mobile.model.db.DownloadEntry;
-import org.edx.mobile.module.db.IDatabase;
-import org.edx.mobile.util.AppConstants;
-import org.junit.Test;
-import org.robolectric.Robolectric;
-import org.robolectric.Shadows;
-import org.robolectric.shadows.ShadowActivity;
-
-import static org.assertj.android.api.Assertions.assertThat;
-import static org.assertj.core.api.Java6Assertions.assertThat;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-
-public abstract class BaseVideosDownloadStateActivityTest extends BaseFragmentActivityTest {
-    /**
-     * Method for defining the subclass of {@link BaseVideosDownloadStateActivity}
-     * that is being tested. Should be overridden by subclasses.
-     *
-     * @return The {@link BaseVideosDownloadStateActivity} subclass that is being
-     *         tested
-     */
-    @Override
-    protected Class<? extends BaseVideosDownloadStateActivity> getActivityClass() {
-        return BaseVideosDownloadStateActivity.class;
-    }
-
-    /**
-     * Testing options menu initialization
-     */
-    @Test
-    public void initializeOptionsMenuTest() {
-        BaseVideosDownloadStateActivity activity =
-                Robolectric.buildActivity(getActivityClass())
-                        .withIntent(getIntent()).setup().get();
-        Menu menu = Shadows.shadowOf(activity).getOptionsMenu();
-        assertNotNull(menu);
-        // Can't see any method to confirm action layout source as well
-        MenuItem progressItem = menu.findItem(R.id.download_progress);
-        assertNotNull(progressItem);
-        assertThat(progressItem).hasTitle(activity.getText(R.string.action_settings));
-    }
-
-    /**
-     * Testing download progress menu visibility states and click behaviour
-     * (starting DownloadActivity). Only when both AppConstants.offline_flag
-     * is true and there is a downloading entry in the database, should the
-     * progress bar be visible.
-     */
-    @Test
-    public void downloadProgressViewTest() {
```

Comment:
This test can be used/moved to a separate class that tests the same behavior for CourseTabsDashboardActivity / CourseTabsDashboardFragment.

Reply:
Added its details in following created story.